modules/rosetta/src/main/scala/org/tessellation/rosetta/domain/construction/ConstructionService.scala

```diff
@@ -4,129 +4,183 @@ import cats.data.{EitherT, NonEmptyList, NonEmptySet}
+        EitherT
+          .fromEither(KryoSerializer[F].deserialize[DAGTransaction](hex.toBytes))
+          .leftMap(_ => MalformedTransaction)
+          .flatMap { transaction =>
+            EitherT {
+              signature.publicKey.hexBytes.toPublicKeyByEC
+                .map(pk => SignatureProof(pk.toId, Signature(signature.hexBytes)).asRight[ConstructionError])
+            }.flatMap { proof =>
+              EitherT
+                .fromEither(
+                  KryoSerializer[F]
+                    .serialize(Signed[DAGTransaction](transaction, NonEmptySet.of(proof)))
+                )
+                .leftMap(_ => MalformedTransaction.asInstanceOf[ConstructionError])
+                .map(Hex.fromBytes(_))
+            }
+          }
 
-    def parseUnsignedTransaction(hex: Hex): EitherT[F, ConstructionError, ConstructionParse.ParseResult] =
-      for {
-        unsignedTransaction <- KryoSerializer[F].deserialize[Transaction](hex.toBytes).toEitherT.leftMap(_ => MalformedTransaction)
-        operations <- transactionToOperations(unsignedTransaction).toEitherT
-      } yield ConstructionParse.ParseResult(operations, none)
-
-    def parseTransaction(hex: Hex, isSigned: Boolean): EitherT[F, ConstructionError, ConstructionParse.ParseResult] =
-      if (isSigned) {
-        parseSignedTransaction(hex)
-      } else {
-        parseUnsignedTransaction(hex)
+      def getPayloads(
+        operations: NonEmptyList[Operation],
+        metadataResult: MetadataResult
+      ): EitherT[F, ConstructionError, PayloadsResult] =
+        EitherT(
+          saltGen().map(transactionSalt =>
+            for {
+              (positiveOperation, negativeOperation) <- getPayloadOperations(operations)
+
+              transactionAmount <- Either.fromOption(positiveOperation.amount.value.toTransactionAmount, InvalidPayloadOperations)
+
+              sourceAddress = negativeOperation.account.address
+              transactionFee <-
+                metadataResult.suggestedFee match {
+                  case None      => TransactionFee.zero.asRight
+                  case Some(amt) => Either.fromOption(amt.value.toTransactionFee, InvalidSuggestedFee)
+                }
+
+              unsignedTx =
+                DAGTransaction(
+                  source = sourceAddress,
+                  destination = positiveOperation.account.address,
+                  amount = transactionAmount,
+                  fee = transactionFee,
+                  parent = metadataResult.metadata.lastReference,
+                  salt = transactionSalt
+                )
+
+              serializedTxn <-
+                KryoSerializer[F]
+                  .serialize(unsignedTx)
+                  .leftMap(e => SerializationError(e.getMessage))
+
+              unsignedTxHash <- unsignedTx.hash.leftMap[ConstructionError](_ => MalformedTransaction)
+              toSignBytes = Hex.fromBytes(unsignedTxHash.getBytes)
```

Comment:
It's not a function, so IMO it should be "signedBytes"

Reply:
Done